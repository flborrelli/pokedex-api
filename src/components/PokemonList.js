import React from "react";
import { Segment } from 'semantic-ui-react';

function PokemonList( { getPokemon, showAllPokemons, filteredPokemons } ) {

  return (
    <div className='d-flex justify-content-center mt-5'>
      <div className='pokelist'>
        {
          showAllPokemons ? 

          getPokemon.map(pokemon => {
          return <Segment raised className='segment' key={pokemon.id}>
          <div className='d-flex align-items-center'>
          <img src={pokemon.sprites.front_default} alt="avatar"/>
          <div>{pokemon.name.charAt(0).toUpperCase() + pokemon.name.slice(1)}</div>
          </div>
          </Segment>
          })
          :
          filteredPokemons.map(pokemon => {
          return <Segment raised className='segment' key={pokemon.id}>
          <div className='d-flex align-items-center'>
          <img src={pokemon.sprites.front_default} alt="avatar"/>
          <div>{pokemon.name.charAt(0).toUpperCase() + pokemon.name.slice(1)}</div>
          </div>
          </Segment>
          })
        }
      </div>
      
    </div>
  );
}

export default React.memo(PokemonList);