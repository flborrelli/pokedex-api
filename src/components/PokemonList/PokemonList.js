import React from "react";
import Loader from "../Loader/Loader";
import { Segment } from "semantic-ui-react";

function PokemonList({ getPokemon, showAllPokemons, filteredPokemons, loader }) {
  return (
    <div className="d-flex justify-content-center mt-5 mb-4">
      <div className="pokelist">
        {loader ? (
          <Loader />
        ) : showAllPokemons ? (
          getPokemon.map(pokemon => {
            return (
              <Segment raised className="segment" key={pokemon.id}>
                <div className="d-flex align-items-center">
                  <img src={pokemon.sprites.front_default} alt="avatar" />
                  <div className='ml-3 pokemon-name'>
                    {pokemon.name.charAt(0).toUpperCase() +
                      pokemon.name.slice(1)}
                  </div>
                </div>
              </Segment>
            );
          })
        ) : filteredPokemons.length > 0 ? (
          filteredPokemons.map(pokemon => {
            return (
              <Segment raised className="segment" key={pokemon.id}>
                <div className="d-flex align-items-center">
                  <img src={pokemon.sprites.front_default} alt="avatar" />
                  <div className='ml-3 pokemon-name'>
                    {pokemon.name.charAt(0).toUpperCase() +
                      pokemon.name.slice(1)}
                  </div>
                </div>
              </Segment>
            );
          })
        ) :
          <div className='d-flex flex-column align-items-center no-match-container'>
            <img className='sad-pokemon mb-4' src="/images/sad-pokemon2.png" alt="sad-pokemon"/>
            <h3>No Pokemon found, please try again.</h3>
          </div>
        }
      </div>
    </div>
  );
}

export default React.memo(PokemonList);
