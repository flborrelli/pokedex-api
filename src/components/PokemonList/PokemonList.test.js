import React from "react";
import ReactDOM from "react-dom";
import PokemonList from "./PokemonList";
import { create } from "react-test-renderer";

const filteredPokemons = [];

const getPokemonData = [
  {
    name: "bulbasaur"
  },
  {
    name: "ivysaur"
  },
  {
    name: "venusaur"
  }
];

test("should match with pokemonlist snapshot", () => {
  const component = create(
    <PokemonList
      filteredPokemons={filteredPokemons}
      getPokemonData={getPokemonData}
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
});
