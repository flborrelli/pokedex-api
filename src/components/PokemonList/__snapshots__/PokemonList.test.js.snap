// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with pokemonlist snapshot 1`] = `
<div
  className="d-flex justify-content-center mt-5 mb-4"
>
  <div
    className="pokelist"
  >
    <div
      className="d-flex flex-column align-items-center no-match-container"
    >
      <img
        alt="sad-pokemon"
        className="sad-pokemon mb-4"
        src="/images/sad-pokemon2.png"
      />
      <h3>
        No Pokemon found, please try again.
      </h3>
    </div>
  </div>
</div>
`;

exports[`should match with pokemonlist snapshot 2`] = `
<div
  className="d-flex justify-content-center mt-5 mb-4"
>
  <div
    className="pokelist"
  >
    <div
      className="d-flex flex-column align-items-center no-match-container"
    >
      <img
        alt="sad-pokemon"
        className="sad-pokemon mb-4"
        src="/images/sad-pokemon2.png"
      />
      <h3>
        No Pokemon found, please try again.
      </h3>
    </div>
  </div>
</div>
`;
